hackathon_site/hardware/serializers.py

```diff
@@ -89,6 +89,56 @@ def get_team_code(obj: Order):
         return obj.team.team_code
 
 
+class OrderChangeSerializer(serializers.ModelSerializer):
+    hardware = HardwareSerializer(many=True, read_only=True)
+    team_code = serializers.SerializerMethodField()
+
+    change_options = {
+        "Submitted": ["Cancelled", "Ready for Pickup"],
+        "Ready for Pickup": ["Picked Up"],
+    }
+
+    class Meta:
+        model = Order
+        fields = OrderListSerializer.Meta.fields
+        read_only_fields = (
+            "id",
+            "hardware",
+            "team",
+            "team_code",
+            "created_at",
+            "updated_at",
+        )
+
+    @staticmethod
+    def get_team_code(obj: Order):
+        return obj.team.team_code
+
+    def validate(self, data):
+        current_status = self.instance.status
+
+        if current_status not in self.change_options:
+            raise ValidationError(
+                {"detail": "Cannot change the status for this order."},
```

Comment:
This doesn't need to be a dict with a "detail" key, you can just put the message in directly. The actual output is the same, so no need to change the tests

Reply:
When this is done 
```
raise ValidationError(
                f"Cannot change the status of an order from {current_status} to {data['status']}.",
                code=status.HTTP_400_BAD_REQUEST,
            )
```
it leads to the error detail json to be {"non_field_errors": error_message} instead of {"detail": error_message}

Reply:
let me know how you want to continue

Reply:
Ah right, that brings up a good point - errors thrown in `validate` cause a generic `non_field_errors` key. 

Could you try making this method `validate_status` and see what happens? I think then it will throw a field error specific to the status, which in theory is useful on the frontend.

Sorry for all the round about changes on this, it's been quite some time since I've done serializer validation.

Reply:
ye nw, will change to status